File: js_solutions/JZ30_Biggest_Sum_Of_Array.js
function FindGreatestSumOfSubArray(array)
{
    // write code here
    if(!array.length) {
        return 0;
    }
    let max = array[0], pre = array[0];
    array.forEach((item, index) => {
        if(index >= 1) {
            pre = pre > 0? pre + item : item;
            max = Math.max(max, pre);
        }
    })
    return max;
}

File: java_solutions/JZ30_Biggest_Sum_Of_Array.java
public class Solution {
    public int FindGreatestSumOfSubArray(int[] array) {
        int pre = 0, max = Integer.MIN_VALUE;
        for(int i = 0; i < array.length; i++) {
            pre = pre > 0? pre + array[i]: array[i];
            max = Math.max(max, pre);
        }
        return max;
    }
}